### ISSUE TITLE
[CEO-ACTION] NAS Configuration — 50TB USB Storage on Workhorse

### ISSUE THREAD

**Issue by ronniepinnell** [ronniepinnell/benchsight#2549]
## Setup Steps

- Connect 50TB USB-A drive to Ubuntu Workhorse
- Detect: `lsblk` and `sudo fdisk -l` to find the drive
- If not formatted: `sudo mkfs.ext4 /dev/sdX1` (CAREFUL — confirm correct drive!)
- Create mount point: `sudo mkdir -p /mnt/nas`
- Mount: `sudo mount /dev/sdX1 /mnt/nas`
- Add to /etc/fstab for auto-mount on reboot: `UUID=<uuid> /mnt/nas ext4 defaults,nofail 0 2` (get UUID with `blkid`)

## Directory Structure

```bash
sudo mkdir -p /mnt/nas/{video/raw,video/processed,video/frames,cv/models,cv/training,cv/annotations,archives/r2-cold,archives/logs,backups}
sudo chown -R ronnie:ronnie /mnt/nas
```

## Network Sharing

- Share via Tailscale: `tailscale serve --bg /mnt/nas` OR set up NFS/Samba for other nodes
- Simpler option: Use `scp` or `rsync` over Tailscale SSH for file transfers

## Verification

- [ ] `df -h /mnt/nas` shows ~50TB
- [ ] `ls /mnt/nas/video/raw` exists
- [ ] After reboot: `mount | grep nas` still shows mounted
- [ ] From mothership: `ssh ronnie@workhorse ls /mnt/nas` returns directory listing

## Dependencies
Blocked by: #2546 Workhorse setup (NAS physically attaches to Workhorse).

**Comment by ronniepinnell** [comment]
## Enrichment Audit — M0001 CEO-Action Review

**Spec alignment:** ⚠️ partial vs §3 (Workhorse NAS capability) + §58 (Storage Management — referenced implicitly via R2 cold archive + NAS)

### Additional Tasks (not in current body)
- [ ] **CEO-critical decision:** what filesystem? ext4 (Linux native, can't read on Mac without plugin) vs exFAT (cross-platform, less robust) vs ZFS (snapshots, but needs RAM). Document the decision before `mkfs`
- [ ] **Before `mkfs`: verify drive is empty** (`lsblk -o NAME,SIZE,FSTYPE,MOUNTPOINT /dev/sdX` and `sudo file -s /dev/sdX1`) — wrong drive = 50TB of data loss
- [ ] **UPS recommendation:** 50TB USB-A drive + unexpected power loss = torn ext4 journal. Factory on workhorse must be on UPS (tag this as prerequisite or note the risk)
- [ ] Write retention policy doc: `/mnt/nas/video/raw` (forever), `/mnt/nas/video/frames` (90 days), `/mnt/nas/archives/logs` (180 days), `/mnt/nas/backups` (rolling 30 days)
- [ ] Create `scripts/companion/nas_health.py` — monitors free space, SMART status, mount health → alerts at 80% full
- [ ] Wire NAS free-space to daily briefing (§24) — "NAS: 12.3TB used / 50TB"
- [ ] Add S.M.A.R.T. check: `sudo smartctl -a /dev/sdX` → schedule weekly + alert on `Reallocated_Sector_Ct > 0`
- [ ] Tailscale Serve for NAS: `tailscale serve --bg --tcp 2049 /mnt/nas` (NFS) OR simpler `rsync over ssh` pattern — pick one and document
- [ ] R2 cold-tier sync cron: `rclone sync /mnt/nas/archives/r2-cold r2:benchsight-cold --min-age 30d` weekly

### Verification Tests (acceptance criteria)
1. **Capacity:** `df -h /mnt/nas | awk 'NR==2 {print $2}'` → expected: `>= 45T` (50TB drives advertise, actual is usually 45-46TiB)
2. **Survives reboot:** `sudo reboot` → after boot: `mountpoint /mnt/nas` → exit 0
3. **Write test:** `dd if=/dev/zero of=/mnt/nas/test.bin bs=1M count=100 && rm /mnt/nas/test.bin` → exit 0, ~100MB/s+ for USB 3.x
4. **SMART clean:** `sudo smartctl -H /dev/sdX | grep "overall-health"` → expected: `PASSED`
5. **Cross-node reach:** from mothership: `ssh workhorse ls /mnt/nas/video/raw` → directory listing (not error)
6. **Permission:** `stat -c '%U:%G' /mnt/nas` → `ronnie:ronnie`
7. **Auto-mount on boot:** remove and re-add `/etc/fstab` line; `sudo mount -a` → mount returns, no error

### Expected Outcomes (done means)
- 50TB mounted, survives reboots unattended (`nofail` ensures boot continues even if drive is unplugged)
- Directory tree exists: `video/{raw,processed,frames}`, `cv/{models,training,annotations}`, `archives/{r2-cold,logs}`, `backups`
- NAS health companion running and wired to morning briefing
- All 4 verification boxes in body checked + the 7 above

### Dependencies
- **Blocks:** #2546 (NAS mount is part of Workhorse setup — either merge this into #2546 or keep sequential), CV pipeline (#M0004), video archive strategy
- **Blocked by:** #2546 (Workhorse up), #2550 (Doppler for rclone R2 keys)
- **Config artifacts:** `/etc/fstab` line, `scripts/companion/nas_health.py`, retention policy doc

### Spec Gaps Surfaced
- §3 mentions NAS storage but no retention policy; §58 is referenced but spec body doesn't detail directory structure — recommend adding the directory tree above to §58.
- ZFS vs ext4 decision is not in spec. Worth a §0.D decision entry.

_Audit trail: automated M0001 CEO-action review. Source: GITHUB_FACTORY_SPEC.md §3, §58._

**Comment by ronniepinnell** [comment]
Need to do this interactive. Scared me

**Comment by ronniepinnell** [comment]
Superseded by #2559 — same task with full 8-phase spec including Samba, multi-machine access, and factory integration.
